pkg/sql/catalog/descs/collection.go

```diff
@@ -191,12 +192,17 @@ type Collection struct {
 	// Only used prior to the 20.2 version upgrade.
 	databaseCache *database.Cache
 
-	// schemaCache maps {databaseID, schemaName} -> (schemaID, if exists, otherwise nil).
-	// TODO(sqlexec): replace with leasing system with custom schemas.
+	// tempSchemaCache maps {databaseID, schemaName} -> catalog.ResolvedSchema.
 	// This is currently never cleared, because there should only be unique schemas
 	// being added for each Collection as only temporary schemas can be
 	// made, and you cannot read from other schema caches.
-	schemaCache sync.Map
+	// TODO (lucy): Figure out a way to possibly get rid of this. Usually we only
+	// need to resolve the temp schema for the current session, in which case
+	// we can get it from the SessionData. There are some exceptions, like
+	// dropping a database with temp schemas. #53163 is related.
+	// TODO (lucy): Also, while this exists, consider clearing the cache every so
```

Comment:
The comment that it is never cleared isnt quite true -- we reset this cache when resetting a `descs.Collection`

Reply:
When does this cache come into play? It seems like it'll happen when resolving a name in a temp schema from a session other than the session which owns it. If that's the case, why do we care about it? Do we even need to cache it? Otherwise, if that's not the case and we use it for also finding temp schemas for the session which owns it, then can't we just not go to the KV and check the session data?

Reply:
Yeah, I originally meant for this to cover both cases (in session and out of session), but I guess you're right that we don't have to cache lookups out of the session, since something like DROP DATABASE CASCADE is a one-off event.

We do also have #53163 as a known issue (the session data only stores the temp schema for the current database, and also never gets updated when the current database changes). So the general cache would bypass this problem. On the other hand, we definitely need to fix that bug soon, at which point this consideration will be irrelevant.

I'll just get rid of the cache. I don't think that will make the temp schema/current database situation much more broken than it is now.

Reply:
i agree with andrew that if the cache is only used in drop database cascade then we actually don't need it

Reply:
I removed the cache, RFAL.